limitador/src/storage/mod.rs

```diff
@@ -21,38 +23,230 @@ pub enum Authorization<'c> {
     Limited(&'c Counter), // First counter found over the limits
 }
 
-pub trait Storage: Sync + Send {
-    fn get_namespaces(&self) -> Result<HashSet<Namespace>, StorageErr>;
-    fn add_limit(&self, limit: &Limit) -> Result<(), StorageErr>;
-    fn get_limits(&self, namespace: &Namespace) -> Result<HashSet<Limit>, StorageErr>;
-    fn delete_limit(&self, limit: &Limit) -> Result<(), StorageErr>;
-    fn delete_limits(&self, namespace: &Namespace) -> Result<(), StorageErr>;
+pub struct Storage {
+    limits: RwLock<HashMap<Namespace, HashSet<Limit>>>,
+    counters: Box<dyn CounterStorage>,
+}
+
+pub struct AsyncStorage {
+    limits: RwLock<HashMap<Namespace, HashSet<Limit>>>,
```

Comment:
Nice. HashSet enforces uniqueness for limits, doesn't it?

Reply:
@alexsnaps Looks like we including the limit's name into the hash for a Limit. I think two limits with everything same except name should be same as well right?

Reply:
@eguzki yes… but has it did before, as `Limit` was used as the key in a `HashMap`.
@rahulanand16nov I didn't change anything there really… but I tend to agree with you indeed. `name` doesn't semantically mean anything to the `Limit`, otoh the `max_value` is part of the `PartialEq#eq` impl. which I think is wrong. I'd rather _not_ conflate these changes with this PR tho. *But* I think we should address these issues too!